apps/manager_app/device_authentication.c

```diff
@@ -0,0 +1,274 @@
+
+/**
+ * @brief Helper function to send serial signature to the host.
+ *
+ * @param state_ptr Pointer to state of type device_auth_flow_states_e depicting
+ * the current state of the flow
+ */
+static void send_serial_signature(device_auth_flow_states_e *state_ptr);
+
+/**
+ * @brief This function is the USB handler callback for the device
+ * authentication flow. It handles the further USB requests from the host to
+ * complete the device authenticaiton flow.
+ *
+ * @param usb_evt The USB event which invoked this callback
+ * @param data_ptr Pointer to state of type device_auth_flow_states_e depicting
+ * the current state of the flow
+ */
+static void device_auth_usb_cb(usb_event_t usb_evt,
+                               device_auth_flow_states_e *state_ptr);
+
+/**
+ * @brief This API sends the response for device authentication flow completion
+ * to the host.
+ *
+ */
+static void send_flow_complete(void);
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+static pb_size_t get_request_type(
+    const manager_auth_device_request_t *request) {
+  return request->which_request;
+}
+
+static void send_flow_complete(void) {
+  manager_result_t result =
+      get_manager_result_template(MANAGER_RESULT_AUTH_DEVICE_TAG);
+
+  result.auth_device.which_response =
+      MANAGER_AUTH_DEVICE_RESPONSE_FLOW_COMPLETE_TAG;
+  result.auth_device.flow_complete.dummy_field = '\0';
+
+  uint8_t response[DEVICE_AUTH_RESPONSE_SIZE] = {0};
+  ASSERT(
+      encode_and_send_manager_result(&result, &response[0], sizeof(response)));
+  return;
+}
+
+static void send_serial_signature(device_auth_flow_states_e *state_ptr) {
+  device_auth_flow_states_e state = *state_ptr;
+
+  /* This will be only called if the correct manager request and
+   * device authentication request is received. It is expected to be checked
+   * within device_authentication API */
+
+  if (SIGN_SERIAL_NUM == state) {
+    /* Fetch and encode serial signature, send it to the host and wait for
+     * further USB requests. */
+    manager_result_t result =
+        get_manager_result_template(MANAGER_RESULT_AUTH_DEVICE_TAG);
+    result.auth_device = sign_serial_number();
+
+    uint8_t response[DEVICE_AUTH_RESPONSE_SIZE] = {0};
+    ASSERT(encode_and_send_manager_result(
+        &result, &response[0], sizeof(response)));
+
+    /* Set the next state to SIGN_RANDOM_NUM */
+    *state_ptr = SIGN_RANDOM_NUM;
+  }
+
+  return;
+}
+
+static void device_auth_usb_cb(usb_event_t usb_evt,
+                               device_auth_flow_states_e *state_ptr) {
+  device_auth_flow_states_e state = *state_ptr;
+
+  /* Decode recieved query using protobuf helpers */
+  manager_query_t query = MANAGER_QUERY_INIT_ZERO;
+  if (false == decode_manager_query(usb_evt.p_msg, usb_evt.msg_size, &query)) {
```

Comment:
in addition to decoding status, add `check_manager_request` too.

Reply:
added this check, thanks